homeassistant/components/kira.py

```diff
@@ -0,0 +1,137 @@
+"""
+KIRA interface to receive UDP packets from an IR-IP bridge.
+"""
+# pylint: disable=import-error
+import itertools
+import logging
+import os
+import yaml
+
+import voluptuous as vol
+from voluptuous.error import Error as VoluptuousError
+
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import (
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_SENSORS,
+    CONF_TYPE,
+    EVENT_HOMEASSISTANT_STOP,
+    STATE_UNKNOWN)
+
+
+REQUIREMENTS = ["pykira==0.1.0"]
+
+DOMAIN = 'kira'
+
+RECEIVERS = {}
+MODULES = {}
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = "0.0.0.0"
+DEFAULT_PORT = 65432
+
+CONF_CODE = "code"
+CONF_REPEAT = "repeat"
+CONF_REMOTES = "remotes"
+CONF_SENSOR = "sensor"
+CONF_REMOTE = "remote"
+
+CODES_YAML = '{}_codes.yaml'.format(DOMAIN)
+
+CODE_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_CODE): cv.string,
+    vol.Optional(CONF_TYPE): cv.string,
+    vol.Optional(CONF_DEVICE): cv.string,
+    vol.Optional(CONF_REPEAT): cv.positive_int,
+})
+
+SENSOR_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME, default=DOMAIN):
+        vol.Exclusive(cv.string, "sensors"),
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+REMOTE_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME, default=DOMAIN):
+        vol.Exclusive(cv.string, "remotes"),
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_SENSORS): [SENSOR_SCHEMA],
+        vol.Optional(CONF_REMOTES): [REMOTE_SCHEMA]})
+}, extra=vol.ALLOW_EXTRA)
+
+
+def load_codes(path):
+    codes = []
+    if os.path.exists(path):
+        with open(path) as f:
+            data = yaml.load(f)
+        for code in data:
+            try:
+                codes.append(CODE_SCHEMA(code))
+            except VoluptuousError as exception:
+                # keep going
+                _LOGGER.warning('Kira Code Invalid Data: %s', exception)
+    else:
+        with open(path, 'w') as f:
+            f.write('')
+    return codes
+
+
+def setup(hass, config):
+    """Setup KIRA capability."""
+    import pykira
+
+    sensors = config[DOMAIN].get(CONF_SENSORS, [])
+    remotes = config[DOMAIN].get(CONF_REMOTES, [])
+    if len(sensors) == 0 and len(remotes) == 0:
+        sensors.append({})
+    codes = load_codes(hass.config.path(CODES_YAML))
+
+    global RECEVIERS
```

Comment:
Please don't use `global`. Use `hass.data` instead, it's a dictionary that you can store data in.